chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -118,6 +120,7 @@ extends HasId {
   private[chisel3] val _commands = ArrayBuffer[Command]()
   private[core] val _ids = ArrayBuffer[HasId]()
   Builder.currentModule = Some(this)
+  Builder.whenDepth = 0
```

Comment:
The other part of that argument is that whenDepth belongs with Builder, and for the sake of parallelism (in the grammatical sense) it might make sense to zero it in the same function where you store it and restore it?